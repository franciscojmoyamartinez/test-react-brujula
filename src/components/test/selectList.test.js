import Renderer from 'react-test-renderer';
import SelectList from "./../selectList";
import {render} from '@testing-library/react';

describe("SelectList", () => {
  test('FormControl exist', () => {
    const {getByTestId} = render(<SelectList />)
    expect(getByTestId('FormControl')).toBeInTheDocument()
  });
  test('Select exist', () => {
    const {getByTestId} = render(<SelectList />)
    expect(getByTestId('Select')).toBeInTheDocument()
  });
  test('Button exist for click', () => {
    const {getByTestId} = render(<SelectList />)
    const button = getByTestId('AddItemButton');
    button.click();
  });
  it('should render', () => {
    const tree = Renderer
    .create(<SelectList />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
