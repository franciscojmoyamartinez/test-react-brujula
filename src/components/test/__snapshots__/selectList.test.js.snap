// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList should render 1`] = `
<div
  className="makeStyles-root-19"
>
  <div
    className="MuiFormControl-root makeStyles-formControl-20"
    data-testid="FormControl"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
      htmlFor="select-multiple-native"
    >
      List Items test offer
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-21 MuiInputBase-formControl MuiInput-formControl"
      data-testid="Select"
      onClick={[Function]}
      onDoubleClick={[Function]}
    >
      <select
        aria-invalid={false}
        className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
        disabled={false}
        id="select-multiple-native"
        multiple={true}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value={Array []}
      >
        <option
          value="Item 1"
        >
          Item 1
        </option>
        <option
          value="Item 2"
        >
          Item 2
        </option>
      </select>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-24 makeStyles-divActions-22"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-action-23"
      data-testid="AddItemButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-action-23"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-action-23"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
